### PR TITLE
Holder for Java migration video.

### PR DESCRIPTION
**Pull request by amouat**
## Type of change
New page for Java migration video

### How should this PR be tested?
Please check for any obvious mistakes and that links work.

Also, I'm never sure what to set the weight to (where it appears in the list) -- I'm happy to change that.

### CODE REVIEW

File: content/chainguard/chainguard-images/videos/java-images.md [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+
+Okay, I want to give a quick overview of how to use the Chainguard Java images.
+
+And in particular, I want to show how to port an existing application to use the Chainguard Java images.
+
+So our images are largely equivalent to the existing Java images that you can find on the Docker Hub, such as the Eclipse Temurin ones.
+
+The difference is that we're much more focused on producing minimal images with a low CVE count.
+
+We do build our own JDK and there's a really great blog on the Chainguard site that explains how we do this and how we bootstrap from the really early versions of Java, which I thoroughly recommend checking out and I will link in the notes.
```
Comment (ltagliaferri) [inline content/chainguard/chainguard-images/videos/java-images.md:40]:
```suggestion
We do build our own JDK and there's a [really great blog](https://www.chainguard.dev/unchained/fully-bootstrapping-java-from-source-in-wolfi) on the Chainguard site that explains how we do this and how we bootstrap from the really early versions of Java, which I thoroughly recommend checking out and I will link in the notes.
```

File: content/chainguard/chainguard-images/videos/java-images.md [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+
+So we should be able to build that fairly easily and now I should be able to run it.
+
+That was pretty quick because it was cached already.
+
+If you build it yourself it will take a little bit longer because it will have to download all the various dependencies.
+
+So let's see if I can get this right.
+
+docker run --rm -d --port 8080 to port 8080 on the host.
```
Comment (ltagliaferri) [inline content/chainguard/chainguard-images/videos/java-images.md:62]: I am having trouble escaping right for a suggestion, but what if you break it onto two lines, so you can have the command easy for someone to copy?

```
docker run --rm -d --port 8080 
```

to port 8080 on the host.

File: content/chainguard/chainguard-images/videos/java-images.md [hunk 3]
```diff
@@ -0,0 +1,173 @@
...
+
+Image was called java-maven.
+
+So now hopefully if I do curl localhost://8080/hello I get "Hello world" back.
```
Comment (ltagliaferri) [inline content/chainguard/chainguard-images/videos/java-images.md:66]: You can also potentially do the `curl` command on its own line for easy copy pasta

File: content/chainguard/chainguard-images/videos/java-images.md [hunk 4]
```diff
@@ -0,0 +1,173 @@
...
+
+So it doesn't have all the build tooling associated with the maven image.
+
+Okay let's try building that.
+
+That looks good.
+
+So if I do... well I guess we should prove it still works.
+
+So if I do docker run.
```
Comment (ltagliaferri) [inline content/chainguard/chainguard-images/videos/java-images.md:144]: You can make these commands at the end on their own lines again

File: content/chainguard/chainguard-images/videos/java-images.md [hunk 5]
```diff
@@ -0,0 +1,173 @@
...
+
+Yeah, so zero CVEs but we've reduced the size and number of packages in the image.
+
+So that's quite a big win.
+
+Took a little bit more work to get to the multi-stage build but not a ridiculous amount.
+
+So that's about it.
+
+We've seen how you can use Chainguards maven and JRE images to reduce the size and CVE count in a Java application with relatively little work.
```
Comment (ltagliaferri) [inline content/chainguard/chainguard-images/videos/java-images.md:170]:
```suggestion
We've seen how you can use Chainguard's maven and JRE images to reduce the size and CVE count in a Java application with relatively little work.
```

**Review by ltagliaferri — APPROVED** [review]
A few small suggestions, mostly to make this easier to someone reading and following along with the tutorial itself.

File: content/chainguard/chainguard-images/videos/java-images.md (hunk 3 above)
Comment (amouat) [reply content/chainguard/chainguard-images/videos/java-images.md:66]: I put it in code formatting, I think that should be enough

**Pull request merged**